### DISCUSSION TITLE
OpenAi assistant and SK - conversation

### DISCUSSION THREAD

**Discussion by JP-droidic** [microsoft/semantic-kernel#8815] [Q&A] [answered]
Hi team

Although I had a similar question before, I'm creating a new topic as I need more clarification. According to what I found, we should have one Assistant for each set of functionalities. This raises two questions: 

1. As far as I can see, once an assistant is created, we can't update it's system message. In open AI documentation, there is an API call for an assistant update,  thread update or individual run where you can overwrite the system message (the simple reason we have a new version of the system prompt). How do you do that using SK?

2. We have OpenAIThreadCreationOptions, but we do not have anything similar for Thread update. A very simple UC - user asks a question, LLM replies, the user now wants to upload a document; we can't attach this document to the ongoing Thread. The only way, as far as I can see, is to create a new thread, copy chat history from the original, attach a file(s) and then continue with the new thread.  Any other suggestions?

Thanks!

**Comment by evchaki** [comment, marked as answer]
@JP-droidic - thanks for bringing this up. 

For #1 - you can update the system message for Python - we will double check on .NET and if it not there we will update it.  However, as a best practice you should create a new agent if you want them to do a certain thing vs switching out system messages each time.  This will make it easier for testing @crickman 
For #2 - this is a gap we will close for .NET and for Python.

**Reply by JP-droidic** [reply]
Thank you @evchaki, 

As far as I'm aware it's not possible in .net. It makes sense to creating a new agent 

For two, there is indeed a big gap. Can you please let me know once you create an issue the ID of it so I can monitor progress

Thanks
